### ISSUE TITLE
Move docs to gitbook

### ISSUE THREAD

**Issue by kulakowka** [ianstormtaylor/slate#257]
It is very uncomfortable to read the documentation on github.

Maybe we could create documentation using [gitbook](https://www.gitbook.com). 

It's pretty convenient. For example look at [feathers.js documentation](http://docs.feathersjs.com/).

I could be of assistance in this task.

Or is there another way to read the documentation?

**Comment by SamyPesse** [comment]
Haha I must say that I totally agree ;)

The plus of GitBook are:
- Sidebar to improve navigation between files (right now we have to switch between a lot of browser tabs to navigate the Slate's documentation)
- Search made easier
- Integration with GitHub: built on all commits and keep the documentation in the `docs` folder.

Happy to help on this, we have great improvements coming soon for API documentation (and GitBook Editor will soon be using Slate 🎉 ), and we'd love to learn more how GitBook can help improve documentation of open source project such as Slate.

**Comment by ianstormtaylor** [comment]
Nice! I'm definitely down. I'll be back on Monday but if anyone wants to implement before then let me know! (Haven't looked into how it works yet)

**Comment by kulakowka** [comment]
I started it: https://kulakowka.gitbooks.io/slate/content/

**Comment by kulakowka** [comment]
Done: https://kulakowka.gitbooks.io/slate/content/

**Comment by kulakowka** [comment]
@ianstormtaylor I can add you as the owner of the book.

**Comment by SamyPesse** [comment]
@kulakowka I'm not sure having the documentation separated from the source code is a good idea; I think we should keep the documentation in the `docs` folder and use GitHub as a single source of truth (PR done to the repository will update the documentation, etc). 

It can be done by adding 3 files:

- `docs/README.md`: describe slate and the documentation
- `docs/SUMMARY.md`
- `book.json` with `{ "root": "./docs" }`

Then a book sync to GitHub can be created on GitBook.com.

**Comment by kulakowka** [comment]
I agree. But I do not have access to the repository to do this. so I copied the documentation with minor corrections.

@SamyPesse Ultimately, we need to do as you have said.

**Comment by ianstormtaylor** [comment]
Hey @kulakowka thanks for looking into it! I didn't realize it would need to be hosted on a specific user's account, or I wouldn't have asked, sorry about that.

I've just tried setting it up, but ran into an error when linking to the GitHub repository, so waiting to hear back on those before we can use it.

**Comment by SamyPesse** [comment]
@ianstormtaylor What is the status of the webhook shown at https://www.gitbook.com/book/ianstormtaylor/slate/settings/github ?

it should be like:

![screen shot 2016-08-23 at 15 26 06](https://cloud.githubusercontent.com/assets/845425/17893759/244d9f9c-6947-11e6-8442-6c69a23cb122.png)

If that's the case, the book will be updated once a new commit is created on the GitHub repository.

**Comment by ianstormtaylor** [comment]
@SamyPesse this is what I see in the settings page:

![image](https://cloud.githubusercontent.com/assets/311752/17902328/bf2ec7ce-691b-11e6-8afa-eab474cf89d0.png)

Do I need to merge your PR with `book.json` first? I couldn't find any good documentation that walked me through why/how to add `book.json`...

**Comment by SamyPesse** [comment]
@ianstormtaylor Ok I'm working on fixing this bug (Sorry, we release an important release yesterday, which introduced a few bugs).

Documentation for the `book.json` can be found at http://toolchain.gitbook.com/config.html

Delete the book on GitBook.com, then merge the PR, and finally create a new book by selecting "GitHub":

<img width="842" alt="out" src="https://cloud.githubusercontent.com/assets/845425/17903365/8d5d9806-696b-11e6-93ae-0d7beeb6903e.png">

Basically it'll importe the GitHub repository, then it'll add a webhook to GitHub, which will trigger an update on GitBook.com everytime the GitHub repository is updated.

**Comment by ianstormtaylor** [comment]
Ah that would do it. I didn't notice the down arrow that reveals the "GitHub" option, so I had been going with "API Documentation" since that seemed best.

**Comment by ianstormtaylor** [comment]
Okay, this is done now! Thanks @SamyPesse for all the help!
